pkg/auth/providers/ldap/ldap_client.go

```diff
@@ -73,6 +73,19 @@ func (p *ldapProvider) loginUser(credential *v32.BasicLogin, config *v3.LdapConf
 		fmt.Sprintf("(%v=%v)", ObjectClass, config.UserObjectClass),
 		operationalAttrList, nil)
 	opResult, err := lConn.Search(searchOpRequest)
+
+	// If no object is found, try again using SA
+	if err != nil && ldapv3.IsErrorWithCode(err, ldapv3.LDAPResultNoSuchObject) {
+		// Re-bind SA
+		err = ldap.AuthenticateServiceAccountUser(serviceAccountPassword, serviceAccountUserName, "", lConn)
+		if err != nil {
+			return v3.Principal{}, nil, err
+		}
+
+		// Re-execute the search
+		opResult, err = lConn.Search(searchOpRequest)
```

Comment:
We should add error handling for err returned by the Search

Reply:
The error returned by the L86 is handled and wrapped in L90.